source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,315 @@
+        const auto center2D = estimate2DMarkerCenters(markers);
+
+        // estimating 3d corners and center from 2d data
+        const auto corners3D = estimate3DMarkerPoints(pointCloud, corners2D);
+        const auto center3D = estimate3DMarkerPoints(pointCloud, center2D);
+
+        // extracting origin and calcualting normal vectors for x-, y- and z-axis
+        auto origin = cv::Vec3d{ center3D[0].x, center3D[0].y, center3D[0].z };
+
+        auto xAxis = cv::Vec3d{ corners3D[2].x - corners3D[1].x,
+                                corners3D[2].y - corners3D[1].y,
+                                corners3D[2].z - corners3D[1].z };
+
+        auto yAxis = cv::Vec3d{ corners3D[0].x - corners3D[1].x,
+                                corners3D[0].y - corners3D[1].y,
+                                corners3D[0].z - corners3D[1].z };
+
+        auto u = xAxis / cv::norm(xAxis, cv::NORM_L2);
+        auto v = yAxis / cv::norm(yAxis, cv::NORM_L2);
+        auto normal = u.cross(v);
+        auto unitNormal = normal / cv::norm(normal, cv::NORM_L2);
+
+        auto rotationMatrix = cv::Mat(3, 3, CV_64F);
+        for(int i = 0; i < 3; ++i)
+        {
+            rotationMatrix.at<double>(i, 0) = u[i];
+            rotationMatrix.at<double>(i, 1) = v[i];
+            rotationMatrix.at<double>(i, 2) = unitNormal[i];
+        }
+
+        auto transform = transformationMatrix(rotationMatrix, origin);
+
+        std::cout << "Displaying marker's pose" << std::endl;
+        displayMarkerPose(pointCloud, corners2D, center2D);
+
+        return transform;
+    }
+
+    Zivid::Matrix4x4 invertAffineTransformation(const Zivid::Matrix4x4 &transform)
+    {
+        auto rotation = cv::Mat(3, 3, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                rotation.at<float>(i, j) = transform(i, j);
+            }
+        }
+
+        auto translation = cv::Mat(3, 1, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            translation.at<float>(i, 0) = transform(i, 3);
+        }
+
+        cv::Mat newRotation = rotation.inv();
+        cv::Mat newTranslation = -newRotation * translation;
+
+        auto cvInverted = cv::Mat(4, 4, CV_32F, 0.0);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                cvInverted.at<float>(i, j) = newRotation.at<float>(i, j);
+            }
+        }
+        for(int i = 0; i < 3; ++i)
+        {
+            cvInverted.at<float>(i, 3) = newTranslation.at<float>(i, 0);
+        }
+        cvInverted.at<float>(3, 3) = 1.0;
+
+        auto *ptr = &cvInverted.at<float>(0, 0);
+        auto invertedTransform = Zivid::Matrix4x4(ptr, ptr + 16);
+
+        return invertedTransform;
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        const auto dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ArucoSample.zdf";
```

Comment:
I think we can improve names for "ArucoSample.zdf" and "transformedArucoSample.zdf"
- 'Sample': this is not needed imo - what value does it give?
- We should indicate somehow the origin of the point cloud: e.g. ArUcoInCameraOrigin and ArUcoInArUcoOrigin? (the latter maybe doesn't sound well...), or PointCloudInCameraOrigin and PointCloudInMarkerOrigin?

Reply:
The name of the file is not finsihed. I just needed to call it something. So good that you start a discussion on the naming of the file!

Reply:
I am implementing this in the other PR: #165